### PR TITLE
Clear obsolete pending lemmas in arithmetic

### PR DESCRIPTION
**Pull request by nafur**
Arithmetic generally generates nonlinear lemmas as full effort checks, but only sends them out in the subsequent last call effort check. If this one does not happen, the next full effort check still has pending lemmas that may not only be irrelevant now, but possibly carry proofs that are already out of scope.
This PR makes sure that such lemmas are always pruned.
Fixes cvc5/cvc5-projects#465

### CODE REVIEW

File: src/theory/arith/theory_arith.cpp [hunk 1]
```diff
@@ -164,19 +164,25 @@ void TheoryArith::postCheck(Effort level)
 {
   d_im.reset();
   Trace("arith-check") << "TheoryArith::postCheck " << level << std::endl;
+  if (Theory::fullEffort(level))
```
Comment (ajreynol) [inline src/theory/arith/theory_arith.cpp:167]: Can this be moved to line 202?

File: test/regress/regress0/nl/proj-issue-465-asan-proofs.smt2 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+; COMMAND-LINE: -i -q
```
Comment (ajreynol) [inline test/regress/regress0/nl/proj-issue-465-asan-proofs.smt2:1]:
```suggestion
; COMMAND-LINE: -i -q --produce-proofs
```

File: src/theory/arith/theory_arith.cpp (hunk 1 above)
Comment (nafur) [reply src/theory/arith/theory_arith.cpp:167]: I don't think so. Note that there are multiple ways to return between line 167 and line 202.

**Comment by 4tXJ7f** [comment]
FWIW, I have investigated the underlying issue of #8253/https://github.com/cvc5/cvc5-projects/issues/377 and I also ended up finding that a stale buffered lemma was to blame (the lemma was buffered, cvc5 returned from the check, the context was popped, and the proof for the lemma was freed).

File: test/regress/regress0/use_approx/specsharp-WindowsCard.15.RTE.Terminate_System.Int32_approx.smt2 [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ; REQUIRES: glpk
-; COMMAND-LINE: --use-approx
+; COMMAND-LINE: --use-approx --no-check-unsat-cores
```
Comment (ajreynol) [inline test/regress/regress0/use_approx/specsharp-WindowsCard.15.RTE.Terminate_System.Int32_approx.smt2:2]: Use `; DISABLE-TESTER: unsat-core`

File: test/regress/regress1/nl/arrowsmith-050317.smt2 [hunk 4]
```diff
@@ -1,4 +1,4 @@
-; COMMAND-LINE: --nl-ext=full
+; COMMAND-LINE: --nl-ext=full --no-check-unsat-cores
```
Comment (ajreynol) [inline test/regress/regress1/nl/arrowsmith-050317.smt2:1]: Ditto.

**Review by ajreynol — CHANGES_REQUESTED** [review]

**Review by ajreynol — APPROVED** [review]

**Pull request merged**